x-pack/plugins/maps/public/classes/sources/es_search_source/load_index_settings.ts

```diff
@@ -4,28 +4,33 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { i18n } from '@kbn/i18n';
 import {
   DEFAULT_MAX_RESULT_WINDOW,
   DEFAULT_MAX_INNER_RESULT_WINDOW,
   INDEX_SETTINGS_API_PATH,
 } from '../../../../common/constants';
 import { getHttp, getToasts } from '../../../kibana_services';
-import { i18n } from '@kbn/i18n';
 
 let toastDisplayed = false;
-const indexSettings = new Map();
+const indexSettings = new Map<string, INDEX_SETTINGS | Promise<INDEX_SETTINGS>>();
```

Comment:
Why is `INDEX_SETTINGS | Promise<INDEX_SETTINGS>`? Would this not always be `Promise<INDEX_SETTINGS>`?